File: shared/types/worker.ts
/**
 * A worker verified and onboarded by a labour association/federation.
 * Workers do not self-register — credentials are issued by an admin
 * (see Admin Dashboard, Phase 5).
 *
 * Phase 6A: field names/shape corrected to match the actual backend
 * `WorkerPublic` response (`backend/app/schemas/worker.py`) exactly —
 * this interface previously described fields (`name`, `phoneNumber` as
 * non-nullable, `skills`, `pinCode`, `isAvailable`) that do not exist on
 * the real `Worker` model. Nothing in `mobile/` or `admin/` consumed
 * this interface at the time of this correction, so this is a type-only
 * change with no behavior impact.
 */
export type WorkerStatus = 'ACTIVE' | 'INACTIVE';

export interface Worker {
  id: string;
  accountId: string;
  associationId: string;
  workerCode: string;
  fullName: string;
  /** Nullable on the backend — a worker's phone is not always recorded. */
  phoneNumber: string | null;
  status: WorkerStatus;
  address: string;
  pincode: string;
  /** 0–5 rating, as a plain number (backend returns a decimal). */
  rating: number;
  totalJobsCompleted: number;
  createdAt: string;
  updatedAt: string;
  // Deliberately no `skills` field: skills are modeled as a separate
  // WorkerSkill join table on the backend, not an array on Worker
  // itself, and no route currently returns them embedded here.
}

File: backend/app/schemas/worker.py
"""
Typed response models for the Phase 5E-D worker-related read routes:
`GET /associations/me/workers`, `GET /workers/me`, `GET /federation/me/workers`.

Exposes ONLY fields that exist on the current, locked Phase 5C `Worker`
model. Per the Phase 5E-D reconciliation: `Worker.address`, `.pincode`,
`.rating`, and `.total_jobs_completed` do NOT exist in the current
database schema and are deliberately NOT included here — adding them
(worker location/statistics data for future matching) is explicitly out
of scope for this phase.
"""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import WorkerStatus


class WorkerPublic(BaseModel):
    """Safe, public view of a `Worker`. Never includes `password_hash` or Account internals."""

    model_config = ConfigDict(from_attributes=True, populate_by_name=True)

    id: UUID
    account_id: UUID = Field(alias="accountId")
    association_id: UUID = Field(alias="associationId")
    worker_code: str = Field(alias="workerCode")
    full_name: str = Field(alias="fullName")
    phone: str | None = Field(alias="phoneNumber")
    status: WorkerStatus
    created_at: datetime = Field(alias="createdAt")
    updated_at: datetime = Field(alias="updatedAt")


class WorkerListResponse(BaseModel):
    """Paginated envelope for a worker list."""

    model_config = ConfigDict(populate_by_name=True)

    items: list[WorkerPublic]
    page: int
    page_size: int = Field(alias="pageSize")
    total: int
